### PR TITLE
Add next billing cycle to /me endpoint

### PR DESCRIPTION
**Pull request by gonzaloriestra**
Related to https://github.com/CartoDB/cartodb/issues/14463

Added a new field `next_billing_period`, with the date of the next billing cycle (when the data services quotas are reset) to the response of `/me` endpoint.

I've also removed some old logic that allowed to calculate the last billing cycle when the user hasn't a `period_end_date`, as I've checked that all of the users in production has it, and it also has a default database value when creating new users.

### CODE REVIEW

File: spec/models/carto/helpers/billing_cycle_spec.rb [hunk 1]
```diff
@@ -94,4 +68,70 @@ class MyDummyUser < OpenStruct
...
+      { today: '2017-03-01', period_end_date: '2016-10-01', expected: '2017-04-01' },
+      { today: '2017-03-10', period_end_date: '2017-02-01', expected: '2017-04-01' },
+      { today: '2017-02-01', period_end_date: '2017-02-01', expected: '2017-03-01' },
+      { today: '2017-01-18', period_end_date: '2016-10-15', expected: '2017-02-15' },
+      { today: '2017-03-12', period_end_date: '2015-07-04', expected: '2017-04-04' }
+    ].each do |example|
+      today = example[:today]
+      period_end_date = example[:period_end_date]
+      expected = example[:expected]
+      it "returns the next month if period_end_date.day >= today.day" \
```
Comment (alrocar) [inline spec/models/carto/helpers/billing_cycle_spec.rb]:
```suggestion
       it "returns the next month if period_end_date.day <= today.day" \
```

Comment (gonzaloriestra) [reply spec/models/carto/helpers/billing_cycle_spec.rb]: So nice the GitHub's suggestion functionality!

File: app/models/carto/helpers/billing_cycle.rb [hunk 2]
```diff
@@ -3,8 +3,19 @@
 module Carto
   module BillingCycle
     def last_billing_cycle
-      day = period_end_date ? period_end_date.day : 29.days.ago.day
-      date = (day > Date.today.day ? (Date.today - 1.month) : Date.today)
+      day = period_end_date.day
```
[the remark this answers is not in the document]
Comment (alrocar) [reply app/models/carto/helpers/billing_cycle.rb]: yep, I think it's fine 👍

**Comment by alonsogarciapablo** [comment]
I've deployed to staging and verified that `next_billing_period` is present in the response:

image

@gonzaloriestra is there anything else I should be testing?

**Comment by gonzaloriestra** [comment]
@alonsogarciapablo no, that's the only change. Maybe we could also check that it returns the right date for different values of the user's `period_end_date`, but it's well covered with the specs, so I think there's no need. Thanks!

**Comment by alonsogarciapablo** [comment]
:+1: Thanks!

**Pull request merged**